diff --git a/dist/core/routing/rewrite.js b/dist/core/routing/rewrite.js
index 7af06eaa4f51a69d9c24e6b4954388b1389eb8cd..e2b6ece5418c0be995735599d67cdaa0643f3e0f 100644
--- a/dist/core/routing/rewrite.js
+++ b/dist/core/routing/rewrite.js
@@ -59,7 +59,7 @@ function findRouteToRewrite({
     if (route.pattern.test(decodedPathname)) {
       if (route.params && route.params.length !== 0 && route.distURL && route.distURL.length !== 0) {
         if (!route.distURL.find(
-          (url) => url.href.replace(outDir.toString(), "").replace(/(?:\/index\.html|\.html)$/, "") == trimSlashes(decodedPathname)
+          (url) => url.href.replace(outDir.toString(), "").replace(/(?:\/index\.html|\.html)$/, "") == trimSlashes(pathname)
         )) {
           continue;
         }
